### DISCUSSION TITLE
Exception.Data destructuring support

### DISCUSSION THREAD

**Discussion by osudude** [RehanSaeed/Serilog.Exceptions#25] [Q&A]
Our code base will frequently put exception relevant data in Exception.Data (param values, user identity, sql statements, etc).   Does Serilog.Exceptions have any support for destructuring Exception.Data key/values?  Our current logging framework will log these values and was curious how to achieve comparable functionality using serilog.

**Comment by RehanSaeed** [comment]
Where an exception type has no custom properties over and above `System.Exception` or where certain common types of exceptions have been catered for like `System.ArgumentNullException` then these are destructured with zero reflection. Where there are exception types that do not fit into the above category, we use reflection to handle this case.

In both cases above, the `Exception.Data` property is logged.

Hope that helps!
